systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaNodePoolUtils.java

```diff
@@ -0,0 +1,85 @@
+    }
+
+    private static final Logger LOGGER = LogManager.getLogger(PodUtils.class);
+
+    public static KafkaNodePool getKafkaNodePool(String namespaceName, String resourceName) {
+        return KafkaNodePoolResource.kafkaNodePoolClient().inNamespace(namespaceName).withName(resourceName).get();
+    }
+
+    public static List<Integer> getCurrentKafkaNodePoolIds(String namespaceName, String resourceName) {
+        return getKafkaNodePool(namespaceName, resourceName).getStatus().getNodeIds();
+    }
+
+    public static String annotateKafkaNodePool(String namespaceName, String resourceName, String annotationKey, String annotationValue) {
+        LOGGER.info("Annotating KafkaNodePool: {}/{} with annotation: {}", namespaceName, resourceName, annotationKey);
+        return ResourceManager.cmdKubeClient().namespace(namespaceName).execInCurrentNamespace("annotate", "kafkanodepool", resourceName, annotationKey + "=" + annotationValue).out().trim();
+    }
+
+    // This method takes into account all possible annotation formats [0, 1, 2, 10-20, 30].
+    public static String annotateKafkaNodePoolNextNodeIds(String namespaceName, String resourceName, String annotationValue) {
+        return annotateKafkaNodePool(namespaceName, resourceName, Annotations.ANNO_STRIMZI_IO_NEXT_NODE_IDS, annotationValue);
+    }
+
+    // In case we want to set IDs for some reason in list of integers [0, 1, 2, 10, 30] format. This method does not represent all possibilities of this annotation.
+    public static String annotateKafkaNodePoolNextNodeIds(String namespaceName, String resourceName, List<Integer> annotationValue) {
+        return annotateKafkaNodePoolNextNodeIds(namespaceName, resourceName, annotationValue.toString());
+    }
+
+    // This method takes into account all possible annotation formats [0, 1, 2, 10-20, 30].
+    public static String annotateKafkaNodePoolRemoveNodeIds(String namespaceName, String resourceName, String annotationValue) {
+        return annotateKafkaNodePool(namespaceName, resourceName, Annotations.ANNO_STRIMZI_IO_REMOVE_NODE_IDS, annotationValue);
+    }
+
+    // In case we want to set IDs for some reason in list of integers [0, 1, 2, 10, 30] format. This method does not represent all possibilities of this annotation.
+    public static String annotateKafkaNodePoolRemoveNodeIds(String namespaceName, String resourceName, List<Integer> annotationValue) {
+        return annotateKafkaNodePoolRemoveNodeIds(namespaceName, resourceName, annotationValue.toString());
+    }
+
+    public static void scaleKafkaNodePool(String namespaceName, String kafkaNodePoolName, int scaleToReplicas) {
+        LOGGER.info("Scaling KafkaNodePool: {}/{} to {} replicas", namespaceName, kafkaNodePoolName, scaleToReplicas);
+        KafkaNodePoolResource.replaceKafkaNodePoolResourceInSpecificNamespace(kafkaNodePoolName, nodePool -> {
+            nodePool.getSpec().setReplicas(scaleToReplicas);
+        }, namespaceName);
+    }
```

Comment:
This is just maybe my feeling, but I would not encapsulate it in some method like this.
I would normally use it directly in the test, so we know what needs to be changed to "scale the NodePool"

Reply:
used scale() instead